### PR TITLE
Gts 21 gatsby refactor

### PR DESCRIPTION
**Pull request by tnordberg**
This is a cross-project PR, addressing immediate needs re: [GTS-21](https://smm.atlassian.net/browse/GTS-21) but also long term work in [EXTECH](https://smm.atlassian.net/jira/software/projects/EXTECH/boards/33).

This is a refactor/update to make `arduino-base` Gatsby compatible. Here are some things 
- Converted `Example.js` component to `examples/arduino-test-page.js`
- Added running log to test page, so you can see a full history of IPCR communication
- Removed outdated Infestation example messages
- Linking to arduino-base readme
- Simple conditional text coloring to visually indicate successful connection to IPC Renderer
- Updated README to describe implementation with Gatsby

To test, create a Gatsby application, and follow the instructions in this branch's README. Point Stele at your Gatsby site and see if it can successfully connect with the IPC Renderer. 

Test page when IPC Renderer **is not** available: https://d.pr/i/I6NNzp
Test page when IPC Renderer **is** available: https://d.pr/i/XvFJhF

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +92,31 @@ Finally, after following the setup above, you'll need to register an `ipcRendere
 </body>
 ```
 
+If you're using a **Gatsby** application such ass SMM's [app-template](https://github.com/scimusmn/app-template)), you will need to [customize the default `html.js` file](https://www.gatsbyjs.com/docs/custom-html/), and inject a script to make the same reference.
```
Comment (brandonwkipp) [inline README.md]: Might want to spell check this line :)

File: README.md [hunk 2]
```diff
@@ -72,7 +72,10 @@ $ git submodule update --init --force
 After getting the submodule path setup, copy the `arduino-base.ino` file from the `arduino-base` submodule to your `Arduino` directory and rename the new copy to `Arduino.ino`. Use this file as a starting point to start working on your project. Inside the sketch are comments that explain how to use the libraries and how you can use the serial messaging system for parsing and sending messages.
 
 #### Setting up an app
-Finally, after following the setup above, you'll need to register an `ipcRenderer` from `Electron` to allow stele and `arduino-base` to communicate. If you're using the SMM `create-react-app`, this can be accomplished by adding a reference in `public/index.html` file.
+Finally, after following the setup above, you'll need to register an `ipcRenderer` from `Electron` to allow stele and `arduino-base` to communicate.
```
Comment (brandonwkipp) [inline README.md:75]: @tnordberg and I discussed that there should be a little more documentation here to make sure developers understand exactly what to do.

**Review by brandonwkipp — CHANGES_REQUESTED** [review]
I tested this with `app-template` and got it working. I asked @tnordberg to spruce up the docs a little bit to make it more clear what the developer is supposed to for `html.js`

File: README.md (hunk 1 above)
Comment (tnordberg) [reply README.md]: 🍑  whoops, thank you

File: README.md (hunk 2 above)
Comment (tnordberg) [reply README.md:75]: Instructions added.

**Review by brandonwkipp — APPROVED** [review]
LGTM!

**Pull request merged**